src/ICS26Router.sol

```diff
@@ -205,6 +212,23 @@ contract ICS26Router is IICS26Router, IICS26RouterErrors, Ownable, ReentrancyGua
 
         ICS02_CLIENT.getClient(msg_.packet.sourceChannel).membership(membershipMsg);
 
+        // ackPacket will no-op if the packet commitment does not exist
+        try IBC_STORE.deletePacketCommitment(msg_.packet) returns (bytes32 storedCommitment) {
+            if (storedCommitment != ICS24Host.packetCommitmentBytes32(msg_.packet)) {
+                revert IBCPacketCommitmentMismatch(storedCommitment, ICS24Host.packetCommitmentBytes32(msg_.packet));
+            }
+        } catch (bytes memory reason) {
+            if (bytes4(reason) == IICS24HostErrors.IBCPacketCommitmentNotFound.selector) {
+                emit Noop();
+                return; // no-op since the packet commitment already deleted
+            } else {
+                // reverts with the same reason
+                assembly ("memory-safe") {
+                    revert(add(reason, 32), mload(reason))
+                }
+            }
+        }
+
```

Comment:
This call was moved under `membership` call because a no-op would prevent membership from being called, hence proofs from getting cached. (i.e. If the first packet in a batched submit has already been submitted, we still want membership to be called so that other packets on the batch can be processed.